lib/presentation/widgets/app_text_field.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:flutter/material.dart';
+
+enum TextFieldType { normal, password }
+
+/// A customizable text input field widget that can be used to display normal
+/// or password fields.
+class AppTextField extends StatefulWidget {
+  /// The type of text field to display, either normal or password.
+  final TextFieldType type;
+
+  /// The hint text to display in the text field.
+  final String hintText;
+
+  /// The error text to display if the user input is invalid.
+  final String? errorText;
+
+  /// The text controller for the text field.
+  final TextEditingController? controller;
+
+  /// Creates a new [AppTextField] instance.
+  const AppTextField({
+    Key? key,
+    required this.type,
+    required this.hintText,
+    this.controller,
+    this.errorText,
```

Comment:
we need more properties, max lines, validator, etc - that the default text field offers